src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,488 @@
+      make_not_null(&extrinsic_curvature),
+      make_not_null(&inverse_spacetime_metric),
+      make_not_null(&spacetime_unit_normal_vector),
+      make_not_null(&incoming_null_one_form),
+      make_not_null(&outgoing_null_one_form),
+      make_not_null(&incoming_null_vector),
+      make_not_null(&outgoing_null_vector), make_not_null(&projection_ab),
+      make_not_null(&projection_Ab), make_not_null(&projection_AB),
+      make_not_null(&char_projected_rhs_dt_v_psi),
+      make_not_null(&char_projected_rhs_dt_v_zero),
+      make_not_null(&char_projected_rhs_dt_v_minus),
+      make_not_null(&constraint_char_zero_plus),
+      make_not_null(&constraint_char_zero_minus), make_not_null(&char_speeds),
+      face_mesh_velocity, normal_covector, pi, phi, spacetime_metric, coords,
+      gamma1, gamma2, lapse, gauge_source, spacetime_deriv_gauge_source, dt_pi,
+      dt_phi, dt_spacetime_metric, d_pi, d_phi, d_spacetime_metric);
+
+  // Account for moving mesh: char speeds -> cher speeds - n_i v^i_g
+  if (face_mesh_velocity.has_value()) {
+    const auto negative_lambda0 =
+        get(dot_product(normal_covector, *face_mesh_velocity));
+    for (size_t a = 0; a < 4; ++a) {
+      char_speeds.at(a) -= negative_lambda0;
+    }
+  }
+
+  Bjorhus::add_constraint_preserving_terms_to_dt_v_psi(
+      make_not_null(&bc_dt_v_psi), unit_interface_normal_vector,
+      three_index_constraint, char_speeds);
+
+  Bjorhus::add_constraint_preserving_terms_to_dt_v_zero(
+      make_not_null(&bc_dt_v_zero), unit_interface_normal_vector,
+      four_index_constraint, char_speeds);
+
+  if (type_ == detail::ConstraintPreservingBjorhusType::ConstraintPreserving) {
+    Bjorhus::add_constraint_preserving_terms_to_dt_v_minus(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, incoming_null_one_form,
+        outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+        projection_ab, projection_Ab, projection_AB,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, char_speeds);
+  } else if (type_ == detail::ConstraintPreservingBjorhusType::
+                          ConstraintPreservingPhysical) {
+    Bjorhus::add_constraint_preserving_physical_terms_to_dt_v_minus(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, normal_covector,
+        unit_interface_normal_vector, spacetime_unit_normal_vector,
+        incoming_null_one_form, outgoing_null_one_form, incoming_null_vector,
+        outgoing_null_vector, projection_ab, projection_Ab, projection_AB,
+        inverse_spatial_metric, extrinsic_curvature, spacetime_metric,
+        inverse_spacetime_metric, three_index_constraint,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, phi, d_phi, d_pi,
+        char_speeds);
+  } else {
+    ERROR("Failed to set dtVMinus. Input option \""
```

Comment:
did you mean to stream something between the quotes here? Currently this will say `Input option "" Must be one of`.

Reply:
ping

Reply:
Not really - this check should be handled much earlier by `convert_constraint_preserving_bjorhus_type_from_yaml` when using options. In fact, this `else {}` clause should never be reached. I have left it in to complete the `if - else if  - ` chain, but can remove it also.

Reply:
Fixed the error message by removing the `\"\"` here.